lib/rules/require-var-decl-first.js

```diff
@@ -108,111 +108,24 @@
-    } else {
-        firstNode = enclosingScope.body.body[0];
-    }
-
-    if (firstNode.type === 'ExpressionStatement' &&
-        firstNode.hasOwnProperty('expression') === true &&
-        firstNode.expression.hasOwnProperty('value') === true &&
-        firstNode.expression.value === 'use strict') {
-        return firstNode;
-    }
-
-    return null;
-}
-
-function isFirstVarDeclInScope(enclosingScope, varDecl, whitespaceOffsetBeforeVarDecl, commentTokens, file) {
-    var adjustedVarDeclStart = varDecl.range[0];
-    var adjustedScopeStart = enclosingScope.range[0];
-
-    if (enclosingScope.type !== 'Program') {
-        // For function declaration and function expression scope use the top block statement as start
-        // This removes the requirement to offset the function declaration or expression related tokens
-        adjustedScopeStart = enclosingScope.body.range[0];
-        // If enclosing scope node type is Program the range start ignores all comments and whitespace before the
-        // variable declaration
-        adjustedVarDeclStart -= whitespaceOffsetBeforeVarDecl;
-    }
-
-    adjustedVarDeclStart -= getOffsetForBlockStatement(enclosingScope, varDecl, commentTokens, file);
-
-    if (adjustedVarDeclStart === adjustedScopeStart) {
-        return true;
-    }
-
-    return false;
-}
-
-function getCommentOffsetBetweenNodes(previousNode, currentNode, commentTokens, file) {
-    var count;
-    var comment;
-    var commentLength = 0;
-
-    for (count = 0; count < commentTokens.length; count++) {
-        comment = commentTokens[count];
-        if (comment.range[0] >= currentNode.range[1]) {
-            // Stop processing comments that are occurred after current node
-            break;
-        }
-
-        if (previousNode.range[1] < currentNode.range[0] &&
-            comment.range[0] > previousNode.range[0] &&
-            comment.range[1] < previousNode.range[1]) {
-            // Stop processing comments that are within multiple declarators in a single variable declaration
-            // of the previous node and the previousNode is not the parent of currentNode
-            break;
-        }
-
-        if (comment.range[0] > currentNode.range[0] &&
-            comment.range[1] < currentNode.range[1]) {
-            // Stop processing comments that are within multiple declarators in a single variable declaration
-            break;
-        }
-
-        if (previousNode.range[0] >= comment.range[1]) {
-            // Skip comments that occurred before the previous node
-            continue;
+function isVarDeclFirst(varDecl) {
+    var elem = varDecl.previousSibling;
+    while (elem && (elem.isComment || elem.isWhitespace ||
+        elem.type === 'VariableDeclaration' ||
+        elem.type === 'Punctuator' ||
+        elem.type === 'ExpressionStatement' && elem.expression.value === 'use strict')) {
+        // We should goes up from blocks for var declaration.
+        if (elem.type === 'Punctuator' &&
```

Comment:
I'm not sure we even need to allow `var`s inside blocks. I mean, if there is `var` — it should be placed FIRST in scope, while scope for `var` is some kind of function node or the program.

Fixed in the second commit. Please check it out!